eng/pipelines/templates/steps/split-tests.yml

```diff
@@ -0,0 +1,55 @@
+parameters:
+- name: OutputFolder
+  type: string
+  default: "projects"
+- name: numOfTestsPerJob
+  type: number
+  default: 20
+- name: OutputProjectFilePath
+  type: string
+  default: "$(System.DefaultWorkingDirectory)/projects.txt"
+- name: TestTargetDependency
+  type: string
+  default: "not-specific"
+- name: ProjectListOverrideFile
+  type: string
+  default: "ProjectListOverrideFile"
+steps:
+  - pwsh: |
+      dotnet build /t:ProjectDependsOn ./eng/service.proj `
+        /p:ServiceDirectory=* /p:TestTargetDependency="${{parameters.TestTargetDependency}}" `
+        /p:IncludeSrc=false /p:IncludeStress=false /p:IncludeSamples=false  `
+        /p:IncludePerf=false /p:RunApiCompat=false `
+        /p:InheritDocEnabled=false /p:BuildProjectReferences=false `
+        /p:BuildInParallel=true /p:OutputProjectFilePath="${{parameters.OutputProjectFilePath}}"
+    displayName: Find all test dependencies
+
+  - task: PowerShell@2
+    inputs:
+      targetType: filePath
+      filePath: eng/scripts/Split-Tests-Xml.ps1
+      arguments: >-
+        -ProjectFilePath ${{ parameters.OutputProjectFilePath }}
+        -OutputFolder ${{ parameters.OutputFolder }}
+        -numOfTestsPerJob ${{ parameters.numOfTestsPerJob }}
+      ignoreLASTEXITCODE: true
+      pwsh: true
+    displayName: Split testing packages into different files
+
+  - pwsh: |
+      $platformJson = Get-Content "./eng/pipelines/templates/stages/platform-matrix.json" | ConvertFrom-Json
+      $projectFiles = Get-ChildItem "${{ parameters.OutputFolder }}"
+      $platformJson.matrix | Add-Member -Name "${{parameters.ProjectListOverrideFile}}" -value $projectFiles.Name -MemberType NoteProperty
+      $platformJson | ConvertTo-Json  -Depth 9 | Out-File "./eng/pipelines/templates/stages/platform-matrix.json"
+      Copy-Item -Path "./eng/pipelines/templates/stages/platform-matrix.json" -Destination "${{ parameters.OutputFolder }}"
+    displayName: Put the file in matrix json
+  
+  - task: CopyFiles@2
+    inputs:
+      sourceFolder: "${{ parameters.OutputFolder }}"
```

Comment:
Should we just make OutputFilder be the ArtifactStagingDirectory to avoid the need for extra copying?